benches/benchsuite/src/lib.rs

```diff
@@ -127,9 +127,12 @@ impl Fixtures {
             fs::create_dir_all(&index).unwrap();
             git("init --bare");
             git("remote add origin https://github.com/rust-lang/crates.io-index-archive");
+            // git 2.48.0 changed the behavior of setting HEAD, so let's just
```

Comment:
Thanks. Tested it locally and this fixes the issue.

However, I don't understand it. On my machine git is 2.47.0, and still failed. Maybe switching from `main` to `master` is enough?

Reply:
Do you have `init.defaultBranch` set? That will also cause this problem.

Reply:
I pushed an update to drop the explicit ref update, and instead use `--initial-branch` which essentially does the same thing.

Reply:
Confirmed that only with both `symbolic-ref HEAD refs/heads/main` and `branch -f main FETCH_HEAD` it works, regardless what is  set in `init.defaultBranch`.